pkg/controller/registry/reconciler/grpc.go

```diff
@@ -168,6 +169,78 @@ func (c *GrpcRegistryReconciler) ensureService(source grpcCatalogSourceDecorator
 	return err
 }
 
+// updateRegistryPodByDigest is an internal method that verifies the latest catalog source by comparing image digests.
+// If the digests do not match (i.e. there is a new version of the catalog source) then overwrite the catalog source pod.
+// This is done to only have the container runtime (CRI-O) talk to the container registry.
+func (c *GrpcRegistryReconciler) updateRegistryPodByDigest(source grpcCatalogSourceDecorator) bool {
+	podDigestChan := make(chan string, 1)
+	lastCheckedTimestamp := time.Time{}
+	interval := source.Spec.Poll.Interval.Duration
+
+	// check poll value is not zero (default poll value)
+	// if polling interval is zero polling will not be done
+	if interval == time.Duration(0) {
+		return false
+	}
+
+	// check digest every poll interval
+	if time.Now().After(lastCheckedTimestamp.Add(interval)) &&
+		source.CreationTimestamp.Add(interval).Before(time.Now()) {
+		lastCheckedTimestamp = time.Now()
+		go c.getPodDigest(source, podDigestChan)
+	}
+
+	select {
```

Comment:
I don't think this will have the desired effect - select won't block and wait; there's a default case that will immediately return (and once this function returns, nothing is reading from the channel any more).

Reply:
Yes, I see what you mean - my original idea was since `getPodDigest` can potentially take a long time this function would exit and on the next reconcile loop check the channel again for a value. But since I declare the channel within the scope of the function its a local variable that gets GC'ed. I think if I declare the `podDigestChan` at a higher level this idea makes more sense.

Reply:
I'm thinking about adding that channel to the `GrpcRegistryReconciler` struct itself